src/pages/ProfilesExplorerView/components/SceneExploreServiceLabels/components/SceneGroupByLabels/domain/actions/IncludeExcludeAction/IncludeExcludeAction.tsx

```diff
@@ -0,0 +1,71 @@
+import { AdHocVariableFilter } from '@grafana/data';
+import { SceneComponentProps, sceneGraph, SceneObjectBase, SceneObjectState } from '@grafana/scenes';
+import React, { useMemo } from 'react';
+
+import { FiltersVariable } from '../../../../../../../domain/variables/FiltersVariable/FiltersVariable';
+import { GridItemData } from '../../../../../../SceneByVariableRepeaterGrid/types/GridItemData';
+import { EventClearLabelFromFilters } from '../../events/EventClearLabelFromFilters';
+import { EventExcludeLabelFromFilters } from '../../events/EventExcludeLabelFromFilters';
+import { EventIncludeLabelInFilters } from '../../events/EventIncludeLabelInFilters';
+import { FilterButtons } from './ui/FilterButtons';
+
+export interface IncludeExcludeActionState extends SceneObjectState {
+  item: GridItemData;
+}
+
+type Status = 'included' | 'excluded' | 'clear';
+
+export class IncludeExcludeAction extends SceneObjectBase<IncludeExcludeActionState> {
+  constructor({ item }: IncludeExcludeActionState) {
+    super({ item });
+  }
+
+  getStatus(filters: AdHocVariableFilter[]) {
+    let status: Status = 'clear';
+
+    const { key, value } = this.state.item.queryRunnerParams.filters![0];
+
+    filters
+      .filter((f) => f.key === key && (f.operator === '=~' || f.operator === '!~'))
+      .some((f) => {
+        if (!f.value.split('|').includes(value)) {
```

Comment:
**note**: Probably gonna work in 99% of cases but it may lead to weird edge-case scenarios, e.g.

https://github.com/user-attachments/assets/7566ff98-0d97-46a9-8fb4-c0273b3c80a9

Reply:
Good catch, I guess the query builder should strip the `()`in this case